src/main/java/org/broadinstitute/hellbender/tools/walkers/rnaseq/OverhangFixingManager.java

```diff
@@ -149,43 +173,48 @@ public void addSplicePosition(final String contig, final int start, final int en
      */
-    public void addRead(final GATKRead read) {
-        Utils.nonNull(read, "read added to manager is null, which is not allowed");
+    public void addReadGroup(final List<GATKRead> readGroup) {
+        Utils.nonNull(readGroup, "readGroup added to manager is null, which is not allowed");
+        Utils.nonEmpty(readGroup, "readGroup added to manager is empty, which is not allowed");
 
         // if the new read is on a different contig or we have too many reads, then we need to flush the queue and clear the map
         final boolean tooManyReads = getNReadsInQueue() >= MAX_RECORDS_IN_MEMORY;
-        final boolean encounteredNewContig = getNReadsInQueue() > 0 && !waitingReads.peek().read.isUnmapped() && !waitingReads.peek().read.getContig().equals(read.getContig());
+        GATKRead topRead = ((getNReadsInQueue()>0)? waitingReadGroups.peek().get(0).read : null);
+        GATKRead firstNewGroup = readGroup.get(0);
+        final boolean encounteredNewContig = getNReadsInQueue() > 0
+                && !topRead.isUnmapped()
+                && !firstNewGroup.isUnmapped()
+                && !topRead.getContig().equals(firstNewGroup.getContig());
 
         if ( tooManyReads || encounteredNewContig ) {
             if ( DEBUG ) {
-                logger.warn("Flushing queue on " + (tooManyReads ? "too many reads" : ("move to new contig: " + read.getContig() + " from " + waitingReads.peek().read.getContig())) + " at " + read.getStart());
+                logger.warn("Flushing queue on " + (tooManyReads ? "too many reads" : ("move to new contig: " + firstNewGroup.getContig() + " from " + topRead.getContig())) + " at " + firstNewGroup.getStart());
             }
 
             final int targetQueueSize = encounteredNewContig ? 0 : MAX_RECORDS_IN_MEMORY / 2;
-
-            // write the required number of waiting reads to disk
-            while ( getNReadsInQueue() > targetQueueSize ) {
-                writer.addRead(waitingReads.poll().read);
-            }
+            writeReads(targetQueueSize);
         }
 
-        final SplitRead splitRead = new SplitRead(read);
+        List<SplitRead> newReadGroup = readGroup.stream().map(SplitRead::new).collect(Collectors.toList());
 
-        // fix overhangs, as needed
+        // Check every stored read for an overhang with the new splice
         for ( final Splice splice : splices) {
-            fixSplit(splitRead, splice);
+            for (int i = 0; i < newReadGroup.size(); i++) {
+                fixSplit(newReadGroup.get(i), splice);
+            }
         }
+        // add the new reads to the queue
+        waitingReadGroups.add(newReadGroup);
+        waitingReads = waitingReads + newReadGroup.size();
 
-        // add the new read to the queue
-        waitingReads.add(splitRead);
     }
 
     /**
-     * Clean up the list of splices
+     * Clean up the list of splices by removing the lowest half of sequential splices
```

Comment:
Removing half the splices seems really arbitrary.  Is there a more principled way of doing it? Is there ever a way we can tell which splices are not going to be needed again?